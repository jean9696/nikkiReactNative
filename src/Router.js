import React, { Component } from 'react';
import Events from './Events';

export default class Router extends Component {
  constructor(props, context) {
    super(props, context);
    this.state = {
      currentComponent: Events,
    };
  }

  handleChangeComponent = currentComponent => this.setState({ currentComponent });

  render() {
    const CurrentComponent = this.state.currentComponent;
    return <CurrentComponent {...this.props} changeRoute={this.handleChangeComponent} />;
  }
}
