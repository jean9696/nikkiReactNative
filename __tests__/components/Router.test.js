import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Router from '../../src/components/Router';

Enzyme.configure({ adapter: new Adapter() });

const MockComponent = () => <div>Mock</div>;

describe('Testing Router', () => {
  it('should render without crashing', () => {
    const wrapper = shallow(<Router location={[1, 1]} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should change component', () => {
    const wrapper = shallow(<Router location={[1, 1]} />);
    const instance = wrapper.instance();
    instance.handleChangeComponent(MockComponent);
    wrapper.update();
    expect(wrapper.type()).toEqual(MockComponent);
  });
});


