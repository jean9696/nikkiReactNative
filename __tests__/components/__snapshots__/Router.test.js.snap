// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Router should render without crashing 1`] = `
<Events
  changeRoute={[Function]}
  location={
    Array [
      1,
      1,
    ]
  }
/>
`;
